js/src/hooks/useCategories.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import { _x, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useAppSelectDispatch from './useAppSelectDispatch';
+import { CATEGORIES_TO_SHOW_IN_TOOLTIP } from '.~/constants';
+
+const SEPARATOR = _x(
+	', ',
+	'the separator for concatenating the categories where the Attribute mapping rule is applied.',
+	'google-listings-and-ads'
+);
+
+const getDeletedCategoryName = ( categoryId ) => {
+	return sprintf(
+		// translators: %d: number of categories.
+		'Category ID %s (deleted)',
+		[ categoryId ],
+		'google-listings-and-ads'
+	);
+};
+
+/**
+ * Returns the category tree used for Tree Select Control
+ * It also maps the deleted (and previously selected) categories
+ *
+ * @param {Array} [selected] The selected category ID's
```

Comment:
It would be a bit nicer to indicate the type of `selected` since it's compared to other string type values internally.

```suggestion
 * @param {Array<string>} [selected] The selected category ID's
```